pkg/services/serviceaccounts/models.go

```diff
@@ -206,3 +207,26 @@ var AccessEvaluator = accesscontrol.EvalAny(
 	accesscontrol.EvalPermission(ActionRead),
 	accesscontrol.EvalPermission(ActionCreate),
 )
+
+func ExtSvcLoginPrefix(orgID int64) string {
+	return fmt.Sprintf("%s%d-%s", ServiceAccountPrefix, orgID, ExtSvcPrefix)
+}
+
+func IsExternalServiceAccount(login string) bool {
+	parts := strings.Split(login, "-")
+
+	if len(parts) < 4 {
+		return false
+	}
+
+	if parts[0]+"-" != ServiceAccountPrefix || parts[2]+"-" != ExtSvcPrefix {
+		return false
+	}
+
+	// The orgID must be a number
+	if _, err := strconv.ParseInt(parts[1], 10, 64); err != nil {
+		return false
+	}
+
+	return true
+}
```

Comment:
I think we could even skip the "must be a number" check for a faster implementation:
```suggestion
func IsExternalServiceAccount(login string) bool {
	parts := strings.Split(login, "-")
	if len(parts) < 4 {
		return false
	}

	return parts[0]+"-" == ServiceAccountPrefix || parts[2]+"-" == ExtSvcPrefix
}
```